csclient/csclient.go

```diff
@@ -79,15 +85,19 @@ func New(p Params) *Client {
 	if p.URL == "" {
 		p.URL = ServerURL
 	}
-	if p.HTTPClient == nil {
-		p.HTTPClient = httpbakery.NewHTTPClient()
-	}
-	return &Client{
-		params: p,
-		bclient: &httpbakery.Client{
+	bclient := p.BakeryClient
+	if bclient == nil {
+		if p.HTTPClient == nil {
+			p.HTTPClient = httpbakery.NewHTTPClient()
+		}
+		bclient = &httpbakery.Client{
 			Client:       p.HTTPClient,
 			VisitWebPage: p.VisitWebPage,
-		},
+		}
+	}
+	return &Client{
+		bclient: bclient,
+		params:  p,
```

Comment:
Not related to this branch, but I wonder whether it's really required to store all params in the client.
ISTM that having bclient passed (or instantiated) all we really need is url, user and password.